contrib/python/src/python/pants/contrib/python/checks/tasks/checkstyle/checkstyle.py

```diff
@@ -166,30 +173,25 @@ def checkstyle(self, interpreter, sources):
       with self.context.new_workunit(name='pythonstyle',
                                      labels=[WorkUnitLabel.TOOL, WorkUnitLabel.LINT],
                                      cmd=' '.join(checker.cmdline(args))) as workunit:
-        failure_count = checker.run(args=args,
+        return checker.run(args=args,
                                     stdout=workunit.output('stdout'),
                                     stderr=workunit.output('stderr'))
-        return failure_count
 
-  def _constraints_are_py3_compatible(self, constraint_tuple):
+  def _constraints_are_whitelisted(self, constraint_tuple):
     """
     Detect whether a tuple of compatibility constraints
-    contains Python 3-compatible constraints.
-    Remove this method after closing:
-    https://github.com/pantsbuild/pants/issues/5764
+    matches constraints imposed by the merged list of the global
+    constraints from PythonSetup and a user-supplied whitelist.
     """
+    if self._acceptable_interpreter_constraints == []:
+      # The user wants to lint everything.
+      return True
+    num_matches = 0
```

Comment:
In the spirit of your `any` before, how about just:
```python
return all(version.parse(constraint) in self._acceptable_interpreter_constraints
           for constraint in constraint_tuple)
```

And this points out that nowhere here do you rely on a `constraints_tuple`, just an iterable, so consider a less-specific name like `constraints`.

Reply:
Great idea